crates/chainio/clients/avsregistry/src/writer.rs

```diff
@@ -1022,6 +1239,81 @@ mod tests {
         assert!(tx_status);
     }
 
+    #[tokio::test]
+    async fn test_set_add_strategies() {
+        let (_container, http_endpoint, _ws_endpoint) = start_m2_anvil_container().await;
+
+        let private_key = FIRST_PRIVATE_KEY.to_string();
+
+        let avs_writer = build_avs_registry_chain_writer(http_endpoint.clone(), private_key).await;
+
+        let quorum_number = 0;
+        let strategy_params = [StrategyParams {
+            strategy: address!("54945180dB7943c0ed0FEE7EdaB2Bd24620256bc"),
+            multiplier: U96::from(1),
+        }];
+
+        let tx_hash = avs_writer
+            .add_strategies(quorum_number, strategy_params.to_vec())
+            .await
+            .unwrap();
+        let tx_status = wait_transaction(&http_endpoint, tx_hash)
+            .await
+            .unwrap()
+            .status();
+
+        assert!(tx_status);
+
+        let contract_stake_registry =
+            StakeRegistry::new(avs_writer.stake_registry_addr, get_provider(&http_endpoint));
+        assert_eq!(
+            strategy_params[0].strategy,
+            contract_stake_registry
+                .strategyParams(quorum_number, "1".parse().unwrap())
+                .call()
+                .await
+                .unwrap()
+                .strategy
+        );
+    }
+
+    #[tokio::test]
+    async fn test_remove_strategies() {
+        let (_container, http_endpoint, _ws_endpoint) = start_m2_anvil_container().await;
+
+        let private_key = FIRST_PRIVATE_KEY.to_string();
+
+        let avs_writer = build_avs_registry_chain_writer(http_endpoint.clone(), private_key).await;
+
+        let quorum_number = 0;
+        let strategy_params = [StrategyParams {
+            strategy: address!("54945180dB7943c0ed0FEE7EdaB2Bd24620256bc"),
+            multiplier: U96::from(1),
+        }];
+
+        avs_writer
+            .add_strategies(quorum_number, strategy_params.to_vec())
+            .await
+            .unwrap();
+
+        let tx_hash = avs_writer
+            .remove_strategies(quorum_number, [U256::from(1)].to_vec())
+            .await
+            .unwrap();
+        let tx_status = wait_transaction(&http_endpoint, tx_hash)
+            .await
+            .unwrap()
+            .status();
+
+        assert!(tx_status);
+        let contract_stake_registry =
+            StakeRegistry::new(avs_writer.stake_registry_addr, get_provider(&http_endpoint));
+        assert!(contract_stake_registry
+            .strategyParams(quorum_number, "1".parse().unwrap())
+            .call()
+            .await
+            .is_err());
```

Comment:
Let's call `unwrap` here. Asserting `is_err` doesn't give us any leads in case this test fails.

Reply:
the call fails which is expected, but unwrap will lead to test failing , when it should pass

Reply:
what do you have in mind?

Reply:
Oh, sorry. I mistook it for an `is_ok` 😅. We can use an `unwrap_err`

Reply:
looks like `strategyParamsReturn` doesn't implement debug 
should i go to the route of adding another bindings patch ?

Reply:
debug should be by default in forge bind